--- DTEDCapstone_2/MainWindow.xaml.cs	2016-11-04 02:15:52.805958500 -0500
+++ DTEDCapstone/MainWindow.xaml.cs	2016-11-04 02:22:47.787980800 -0500
@@ -8,6 +8,8 @@
 using DTEDCaptsone.Data_Scaling;
 using DTEDCapstone.Plotter;
 
+using DTEDCapstone.Translator;
+
 ///Capstone Project for DTED group
 ///Austin Graham
 ///Brandon Shaw
@@ -84,17 +86,26 @@
             int longiNum = scaledData.GetLength(0);
             int latiNum = scaledData.GetLength(1);
 
-            double[] z = new double[longiNum * latiNum]; // 'z' is elevation
+            int size = longiNum * latiNum;
+            double[] x = new double[size]; // 'x' is distance eastward
+            double[] y = new double[size]; // 'y' is distance northward
+            double[] z = new double[size]; // 'z' is elevation
 
             int counter = 0; // Counter to place elevation in proper order
 
+            CartesianPoint[,] pts = new CartesianData(scaledData).getCartesianData();
+            
             // Iterate from left to right on grid
             for(int i = 0; i < longiNum; ++i)
             {
                 // Iterate from bottom to top on grid
-                for(int j = 0; j < latiNum; ++j, ++counter)
+                for(int j = 0; j < latiNum; ++j)
                 {
-                    z[counter] = scaledData[i, j].Elevation;
+                    //z[counter] = scaledData[i, j].Elevation;
+                    x[counter] = pts[i, j].X;
+                    y[counter] = pts[i, j].Y;
+                    z[counter] = pts[i, j].Z;
+                    counter++;
                 }
             }
 
@@ -103,10 +114,13 @@
             GnuPlot.Set("isosamples 30");
 
             //set the range for the x,y,z axis and plot (using pm3d to map height to color)
-            // GnuPlot.Set("xrange[" + x.Min().ToString() + ":" + x.Max().ToString() + "]", 
-            // "yrange[" + y.Min().ToString() + ":" + y.Max().ToString()  + "]", 
-            // "zrange[" + z.Min().ToString() + ":" + z.Max().ToString() + "]");
-            GnuPlot.SPlot(scaledData.GetLength(1), z, "with lines");
+            /*
+            GnuPlot.Set("xrange[" + x.Min().ToString() + ":" + x.Max().ToString() + "]", 
+             "yrange[" + y.Min().ToString() + ":" + y.Max().ToString()  + "]", 
+             "zrange[" + z.Min().ToString() + ":" + z.Max().ToString() + "]");
+            */
+            //GnuPlot.SPlot(x, y, z); // Don't think this does what I think it does. :/
+            GnuPlot.SPlot(scaledData.GetLength(1), z);
 
             export.IsEnabled = true;
         }
