--- DTEDCapstone_2/Translator/Translate.cs	2016-11-04 02:16:03.153602600 -0500
+++ DTEDCapstone/Translator/Translate.cs	2016-11-04 02:24:24.387486300 -0500
@@ -1,4 +1,5 @@
 ﻿using System.Runtime.InteropServices;
+using System;
 
 /*
  * This class handles the connection to the native GEOTRANS
@@ -16,8 +17,10 @@
     class Translate
     {
         /*Import the needed method from the native .dll*/
+        /* Using regular code to perform standard geospatial transform
         [DllImport("Translator.dll", CallingConvention = CallingConvention.Cdecl)]
         public unsafe static extern string ConvertToCartesian(double lat, double lon, double height, double* x, double* y, double* z);
+        */
 
         /*Construct wrapper method around unsafe method call*/
         public static CartesianPoint Convert(double lat, double lon, double height)
@@ -25,6 +28,13 @@
             //Point to return
             CartesianPoint returnVal;
 
+            height += 6371000; // Height is in meters, add sea-level radius.
+            double frac = Math.Cos(lat);
+            returnVal.X = Math.Cos(lon) * frac * height;
+            returnVal.Y = Math.Sin(lon) * frac * height;
+            returnVal.Z = Math.Sin(lat) * height;
+
+            /*
             //Declare unsafe to make call, use method from native code
             unsafe
             {
@@ -41,6 +51,7 @@
                 returnVal.Y = y;
                 returnVal.Z = z;
             }
+            */
 
             //Return final point
             return returnVal;
